ui/webui/src/apis/storage.js

```diff
@@ -0,0 +1,168 @@
+    return storageClient(address).call(
+        "/org/fedoraproject/Anaconda/Modules/Storage/DiskSelection",
+        "org.freedesktop.DBus.Properties",
+        "GetAll",
+        ["org.fedoraproject.Anaconda.Modules.Storage.DiskSelection"],
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ * @param {string} disk         A device name
+ *
+ * @returns {Promise}           Resolves an object with the device data
+ */
+export const getDeviceData = ({ address, disk }) => {
+    return storageClient(address).call(
+        "/org/fedoraproject/Anaconda/Modules/Storage/DeviceTree",
+        "org.fedoraproject.Anaconda.Modules.Storage.DeviceTree.Viewer",
+        "GetDeviceData", [disk]
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ *
+ * @returns {Promise}           Resolves a list with disk names
+ */
+export const getUsableDisks = ({ address }) => {
+    return storageClient(address).call(
+        "/org/fedoraproject/Anaconda/Modules/Storage/DiskSelection",
+        "org.fedoraproject.Anaconda.Modules.Storage.DiskSelection",
+        "GetUsableDisks", []
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ * @param {string} partitioning DBus path to a partitioning
+ *
+ * @returns {Promise}           Resolves a DBus path to a task
+ */
+export const partitioningConfigureWithTask = ({ address, partitioning }) => {
+    return storageClient(address).call(
+        partitioning,
+        "org.fedoraproject.Anaconda.Modules.Storage.Partitioning",
+        "ConfigureWithTask", []
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ * @param {string} task         DBus path to a task
+ * @param {string} onSuccess    Callback to run after Succeeded signal is received
+ * @param {string} onFail       Callback to run as an error handler
+ *
+ * @returns {Promise}           Resolves a DBus path to a task
+ */
+export const runStorageTask = ({ address, task, onSuccess, onFail }) => {
+    const taskProxy = storageClient(address).proxy(
+        "org.fedoraproject.Anaconda.Task",
+        task
+    );
+    const addEventListeners = () => {
+        taskProxy.addEventListener("Stopped", () => taskProxy.Finish().catch(onFail));
+        taskProxy.addEventListener("Succeeded", onSuccess);
+    };
+    taskProxy.wait(() => {
+        addEventListeners();
+        taskProxy.Start().catch(onFail);
+    });
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ * @param {int} mode            The number of the mode
+ */
+export const setInitializationMode = ({ address, mode }) => {
+    return storageClient(address).call(
+        "/org/fedoraproject/Anaconda/Modules/Storage/DiskInitialization",
+        "org.fedoraproject.Anaconda.Modules.Storage.DiskInitialization",
+        "SetInitializationMode", [mode]
+    );
+};
+
+/**
+ * @param {string} address      Anaconda bus address
+ * @param {int} enabled         True if allowed, otherwise False
```

Comment:
So I guess 0 is False and 1 is True ? Any reason why this is not a boolean value or is this some DBus limitation ?

Reply:
It's my fault, it's boolean. Will update.